### PR TITLE
Feature/security header api testing

### PR DESCRIPTION
**Pull request by sinhaamit1980**

### CODE REVIEW

File: api_tests/helpers/request_helper.rb [hunk 1]
```diff
@@ -6,5 +6,13 @@ def questions(school_pin, pupil_pin)
     self.class.post(BASE_URL+"/api/questions", :body => {:'schoolPin' => school_pin, :'pupilPin' => pupil_pin})
   end
 
+  def admin_home()
+    self.class.get('https://admin-as-dev-mtc.azurewebsites.net/')
+  end
+
+  def spa_home()
```
Comment (activemq) [inline api_tests/helpers/request_helper.rb]: dont need the ()

Comment (activemq) [reply api_tests/helpers/request_helper.rb]: Also would be good to pull in the URLS from somewhere and not have them hardcoded, I dont think they should be pointing to the dev env anyway.

File: api_tests/spec/security_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      expect(response.headers['x-download-options']).to eql 'noopen'
+      expect(response.headers['x-content-type-options']).to eql 'nosniff'
+      expect(response.headers['x-xss-protection']).to eql "1; mode=block"
+      expect(response.headers['content-security-policy']).to eql "default-src 'self'; script-src 'self' 'unsafe-inline' https://www.google-analytics.com; font-src 'self' data:; style-src 'self'; img-src 'self' https://www.google-analytics.com; connect-src 'self'; object-src 'none'; media-src 'none'; child-src 'none'"
+      expect(response.headers['x-content-security-policy']).to eql "default-src 'self'; script-src 'self' 'unsafe-inline' https://www.google-analytics.com; font-src 'self' data:; style-src 'self'; img-src 'self' https://www.google-analytics.com; connect-src 'self'; object-src 'none'; media-src 'none'; child-src 'none'"
+      expect(response.headers['x-webkit-csp']).to eql "default-src 'self'; script-src 'self' 'unsafe-inline' https://www.google-analytics.com; font-src 'self' data:; style-src 'self'; img-src 'self' https://www.google-analytics.com; connect-src 'self'; object-src 'none'; media-src 'none'; child-src 'none'"
```
Comment (GuyHarwood) [inline api_tests/spec/security_spec.rb]: this is webkit specific, please remove

File: api_tests/spec/security_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      response = request_helper.admin_home
+      expect(response.code).to eql 200
+      expect(response.headers['x-dns-prefetch-control']).to eql 'off'
+      expect(response.headers['x-frame-options']).to eql 'SAMEORIGIN'
+      expect(response.headers['strict-transport-security']).to eql "max-age=31536000"
+      expect(response.headers['x-download-options']).to eql 'noopen'
+      expect(response.headers['x-content-type-options']).to eql 'nosniff'
+      expect(response.headers['x-xss-protection']).to eql "1; mode=block"
+      expect(response.headers['content-security-policy']).to eql "default-src 'self'; script-src 'self' 'unsafe-inline' https://www.google-analytics.com; font-src 'self' data:; style-src 'self'; img-src 'self' https://www.google-analytics.com; connect-src 'self'; object-src 'none'; media-src 'none'; child-src 'none'"
+      expect(response.headers['x-content-security-policy']).to eql "default-src 'self'; script-src 'self' 'unsafe-inline' https://www.google-analytics.com; font-src 'self' data:; style-src 'self'; img-src 'self' https://www.google-analytics.com; connect-src 'self'; object-src 'none'; media-src 'none'; child-src 'none'"
```
Comment (GuyHarwood) [inline api_tests/spec/security_spec.rb]: it is generated for a specific browser, not explicit via config
please remove.

File: api_tests/helpers/request_helper.rb (hunk 1 above)
Comment (GuyHarwood) [reply api_tests/helpers/request_helper.rb]: @activemq i agree about storing them in config.  However they do need to point to a containerised instance of the sites that is running on HTTPS.  As some of the headers are only applied by the container software (nginx, for example).  HTTPS redirect is configured specifically for Azure.

**Pull request merged**